pkg/operators/oci-handler/oci.go

```diff
@@ -96,6 +99,20 @@ func (o *ociHandler) InstanceParams() api.Params {
 			Description: "Secret to use when pulling the gadget image",
 			TypeHint:    api.TypeString,
 		},
+		{
+			Key:          verifyImage,
+			Title:        "Verify image",
+			Description:  "Verify image using the provided public key",
+			DefaultValue: "true",
+			TypeHint:     api.TypeBool,
+		},
+		{
+			Key:          publicKey,
+			Title:        "Public key",
+			Description:  "Public key used to verify the image based gadget",
+			DefaultValue: resources.InspektorGadgetPublicKey,
+			TypeHint:     api.TypeString,
+		},
```

Comment:
While I think it's okay to have these in here for testing (and also because it's not 100% cleanly implemented otherwise), those should eventually become `GlobalParams` that are initialized from values set in the deployment/config file instead of being transferred using gRPC from the client (which all these InstanceParams currently are) to ensure safety. So maybe add a TODO for now so we don't forget.

Reply:
Sure!

Operators have two config parts - a global and a per-gadget one (instance params). The global part should be initialized with configuration values from the host itself (e.g. params given to the server binary on the host or a config file on that machine), while all the per-gadget ones are forwarded to the client and can be filled out from there and sent back. So everything concerning security should be coming from the global part or at least be restricted from there (e.g. "a user may override this or that setting" - where user would refer to a client connection).

On `ig`, both params are treated similarly, as it's both client and server/host.

We sadly still don't have support for config files / generic params from the deployment, so the global params don't get used a lot right now. IMHO, this should definitely be handled before removing the experimental flag from image based gadgets.

Reply:
The documentation in docs/devel/hello-world-gadget.md should be updated to explain how to sign third-party gadgets (next to the section "Pushing the gadget image to a container registry").

(this can be in a separate PR if you prefer)

Reply:
> The documentation in docs/devel/hello-world-gadget.md should be updated to explain how to sign third-party gadgets (next to the section "Pushing the gadget image to a container registry").

Do we really want to continue adding more information to the hello-world guide? IMO it's becoming too complex. What about creating a new file under `docs/reference` with this information?

Reply:
> I can drop the last commit if needed. 

I'd prefer to drop it.

Reply:
> There is definitely something I do not understand with the `GlobalParams()`, as their values seem to never change:
> 
Operators' global params will not work because we are always using default values:

https://github.com/inspektor-gadget/inspektor-gadget/blob/162b48b11b1f7bed5995a187bb9b9465f17184d6/pkg/gadget-context/run.go#L44-L51

So, I suggest you to keep the verification params as instance params for this PR, until we properly implement support for setting operators' global params.

Reply:
> > The documentation in docs/devel/hello-world-gadget.md should be updated to explain how to sign third-party gadgets (next to the section "Pushing the gadget image to a container registry").
> 
> Do we really want to continue adding more information to the hello-world guide? IMO it's becoming too complex. What about creating a new file under `docs/reference` with this information?

IMO, we should add one sentence saying that we support signing and then a link to another document where we explain everything about it:

- By default, IG gadgets are signed.
- By default, IG will verify gadgets using the IG pub key.
- Verification ca be skipped using flag.
- People can use and create their own key-pair. Add an example of how to do it.
- And whatever I missed here 🙂